### PR TITLE
Update restore documentation

### LINKED ISSUE #3375: Update documentation of restore procedure (MV and SI issue)
We need to update restore procedure documentation to include information about materialized views and indexes restoration.

Restored table related MV and SI must be drop and recreated after the successful tables restoration.

### PR DESCRIPTION
**Pull request by karol-kokoszka**
Fixes #3375 

---

"Only the Best is Good Enough." - LEGO company motto

Please make sure that:
- Code is split to commits that address a single change
- Commit messages are informative
- Commit titles have module prefix
- Commit titles have issue nr. suffix

### CODE REVIEW

File: docs/source/restore/restore-tables.rst [hunk 1]
```diff
@@ -22,8 +22,34 @@ Prerequisites
...
 * It is important to remember current mode of the `tombstone_gc <https://www.scylladb.com/2022/06/30/preventing-data-resurrection-with-repair-based-tombstone-garbage-collection/>`_ option.
-    In order to ensure data consistency, Scylla Manager alters restored tables options with ``tombstone_gc = {'mode': 'disabled'}``.
-    Right now, it's user responsibility to reset the ``tombstone_gc`` option as a part of the restore tables *follow-up action*.
+   In order to ensure data consistency, Scylla Manager alters restored tables options with ``tombstone_gc = {'mode': 'disabled'}``.
+   Right now, it's user responsibility to reset the ``tombstone_gc`` option as a part of the restore tables *follow-up action*.
+
+* It is strongly advised to drop all materialized views and secondary indexes that are built on top of the table that is about to be restored. They must be recreated after the restore.
+   We do not guarantee that materialized views and secondary indexes will be restored correctly.
+   Materialized views and secondary indexes must be recreated after the table is successfully restored.
+   Please find some useful CQL commands below:
```
Comment (tzach) [inline docs/source/restore/restore-tables.rst:31]: I sugget to reword lines 28-31 to:
".. note::
     Materialized views and secondary indexes must be recreated after the table is successfully restored."

No need to advice or to (not) guarantee. Just list the steps

File: docs/source/restore/restore-tables.rst [hunk 2]
```diff
@@ -34,6 +60,8 @@ Without the repair, the restored content of the tables might not be visible, and
 *ONLY AFTER* repairing restored tables, you should set their `tombstone_gc <https://www.scylladb.com/2022/06/30/preventing-data-resurrection-with-repair-based-tombstone-garbage-collection/>`_ option to desired mode (e.g. previous mode or strongly recommended ``tombstone_gc = {'mode': 'repair'}``).
 Without that, purging tombstones for those tables will remain disabled. This can result in great memory consumption over time, as deleted rows won't ever be removed from database's memory.
 
+The last step is to recreate Materialized Views and Secondary Indexes. To do it, just execute index creation and materialized views creation commands, saved before performing the restore.
```
Comment (tzach) [inline docs/source/restore/restore-tables.rst:63]: Would be useful to add an example for a drop of MV and of SI (in the step above) and for a create of MV and SI.

File: docs/source/restore/restore-tables.rst (hunk 1 above)
Comment (karol-kokoszka) [reply docs/source/restore/restore-tables.rst]: OK, 28-31 is replaced with just "Materialized views and secondary indexes must be recreated after the table is successfully restored." as suggested.

**Comment by karol-kokoszka** [comment]
@tzach Your comments are addressed. Please let me know if there are any other outstanding issues related to restore documentation. 
If there is nothing more, then I'll just merge this PR.

**Pull request merged**